src/Native/System.Native/pal_networking.cpp

```diff
@@ -2013,3 +2018,21 @@ extern "C" Error WaitForSocketEvents(int32_t port, SocketEvent* buffer, int32_t*
 
     return WaitForSocketEventsInner(port, buffer, count);
 }
+
+extern "C" int32_t PlatformSupportsMultipleConnectAttempts()
+{
+#if defined(PLATFORM_SUPPORTS_MULTIPLE_CONNECT_ATTEMPTS)
```

Comment:
@sokket: `#if <symbol>` differs from `#if defined(<symbol>)` in the case that `symbol` is defined to `0`. That said, I don't have a strong preference here, so I'm happy to change this. FWIW, AFAICT we're only using `#if <symbol>` in when `symbol` is guarantee to be defined (e.g. symbols defined by config.h).

@jasonwilliams200OK: we can't detect this feature by the presence of a function, struct, etc., but we could move the `add_definitions` statements to `configure.cmake`.